src/quick-lint-js/error.h

```diff
@@ -747,6 +747,12 @@
       error_missing_operand_for_operator, "E026", { source_code_span where; }, \
       .error(QLJS_TRANSLATABLE("missing operand for operator"), where))        \
                                                                                \
+  QLJS_ERROR_TYPE(                                                             \
+      error_redundant_delete_statement_on_variable, "E086",                    \
+      { source_code_span where; },                                             \
```

Comment:
`where` is a bad variable name. (I know a lot of other error types use it.) Let's pick something better. `delete_operator`? `delete_expression` (if we change it to cover the entire expression, not just the keyword)?